app/assets/javascripts/school_administrator_dashboard/dashboard_components/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -60,9 +54,10 @@ class SchoolDisciplineDashboard extends React.Component {
   }
 
   getChartData(selectedChart) {
+    const filteredEvents = this.filterIncidentDates(this.props.schoolDisciplineEvents);
```

Comment:
@edavidsonsawyer I'm wondering about how this is related to the sorting improvements in the awesome GIFs on the pull request.  This looks like it's adding in another step of filtering, which I thought worked correctly before.  Are these two things related?

Reply:
@kevinrobinson oh good catch! The intention was to move this filter method here so that all the other components just get the filtered data without having to do it themselves. But apparently I forgot to remove the other instances of this filter. 

I updated the PR and commented here about why the filter is being applied and tested with the removed code. This should be clearer and have less repetition now.